From 3d739cd79c28ea5d633a3f64758570b16260df69 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Sat, 8 Oct 2011 13:30:15 +0800
Subject: [PATCH] ENGR00159010 IPUv3:Restore IDMAC_CH_LOCK_EN_1 for resume

This patch restores IDMAC_CH_LOCK_EN_1 register when IPUv3
driver resumes. This avoid the relative issue if setting
IDMAC_CH_LOCK_EN_1 to be zero.

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
(cherry picked from commit fce84cf35dcb338886df8e58f66a7ad1048d2abe)
---
 drivers/mxc/ipu3/ipu_common.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/mxc/ipu3/ipu_common.c b/drivers/mxc/ipu3/ipu_common.c
index 4b12905..1f5ed8b 100644
--- a/drivers/mxc/ipu3/ipu_common.c
+++ b/drivers/mxc/ipu3/ipu_common.c
@@ -2780,6 +2780,9 @@ static int ipu_resume(struct platform_device *pdev)
 
 		/* Set sync refresh channels as high priority */
 		__raw_writel(0x18800001L, IDMAC_CHA_PRI(0));
+
+		/* AXI burst setting for sync refresh channels */
+		__raw_writel(0x003F0000, IDMAC_CH_LOCK_EN_1);
 		clk_disable(g_ipu_clk);
 	}
 	mutex_unlock(&ipu_clk_lock);
-- 
1.5.4.4

